src/services/TelemetryService.ts

```diff
@@ -255,6 +260,23 @@ export class TelemetryService {
         return this.addTelemetryEntry(TelemetryAction.FileClose, {id});
     }
 
+    addSpectralProfileEntry(regionType: CARTA.RegionType, regionId: number, width: number, height: number, depth: number) {
+        switch (regionType) {
+            case CARTA.RegionType.POINT:
+                TelemetryService.Instance.addTelemetryEntry(TelemetryAction.SpectralProfileGeneration, {regionId: regionId, depth});
+                break;
+            case CARTA.RegionType.RECTANGLE:
+            case CARTA.RegionType.POLYGON:
```

Comment:
Looks like polygon is not supported.

Reply:
Do you mean it's not supported for z-profile? Or not supported for sending telemetry data?

Reply:
Not supported for z profile.